src/shared/telemetry/defaultTelemetryClient.ts

```diff
@@ -0,0 +1,66 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import { Credentials, Service } from 'aws-sdk'
+import * as os from 'os'
+import * as vscode from 'vscode'
+import * as constants from '../constants'
+import { ext } from '../extensionGlobals'
+import * as ClientTelemetry from './clienttelemetry'
+import apiConfig = require('./service-2.json')
+import { TelemetryClient } from './telemetryClient'
+import { TelemetryEventArray } from './telemetryEvent'
+
+export class DefaultTelemetryClient implements TelemetryClient {
+    public static readonly DEFAULT_IDENTITY_POOL = 'us-east-1:820fd6d1-95c0-4ca4-bffb-3f01d32da842'
+    public static readonly DEFAULT_TELEMETRY_ENDPOINT = 'https://client-telemetry.us-east-1.amazonaws.com'
+
+    private static readonly PRODUCT_NAME = 'AWS Toolkit For VS Code'
+
+    private constructor(private readonly clientId: string, private readonly client: ClientTelemetry) {}
+
+    /**
+     * Returns failed events
+     * @param batch batch of events
+     */
+    public async postMetrics(batch: TelemetryEventArray): Promise<TelemetryEventArray | undefined> {
+        try {
```

Comment:
Assuming you add a settings field for whether or not to collect data: check the setting value with `../settingsConfiguration`'s `readSetting()` call